crates/walrus-service/src/client/cli/runner.rs

```diff
@@ -627,6 +753,157 @@ impl ClientCommandRunner {
         outputs.print_output(json)
     }
 
+    #[allow(clippy::too_many_arguments)]
+    pub(crate) async fn store_quilt(
+        self,
+        paths: Vec<PathBuf>,
+        blobs: Vec<QuiltBlobInput>,
+        epoch_arg: EpochArg,
+        dry_run: bool,
+        store_optimizations: StoreOptimizations,
+        persistence: BlobPersistence,
+        post_store: PostStoreAction,
+        encoding_type: Option<EncodingType>,
+    ) -> Result<()> {
+        epoch_arg.exactly_one_is_some()?;
+        if encoding_type.is_some_and(|encoding| !encoding.is_supported()) {
+            anyhow::bail!(ClientErrorKind::UnsupportedEncodingType(
+                encoding_type.expect("just checked that option is Some")
+            ));
+        }
```

Comment:
nit: in a follow up PR, I think we should just ignore this input option, or remove this option.